[Bug]: Unable to build with gcc 12.2.1

### What happened?

```shell
/usr/include/bits/string_fortified.h:138:34: error: 'char* __builtin___strncat_chk(char*, const char*, long unsigned int, long unsigned int)' argument 2 declared attribute 'nonstring' [-Werror=stringop-overread]
  138 |   return __builtin___strncat_chk (__dest, __src, __len,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
  139 |                                   __glibc_objsize (__dest));
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/utmp.h:29,
                 from /mgarepo/conky/BUILD/conky-1.18.0/src/users.cc:32:
/usr/include/bits/utmp.h: In function 'int update_users()':
/usr/include/bits/utmp.h:66:8: note: argument 'utmp::ut_user' declared here
   66 |   char ut_user[UT_NAMESIZE]
      |        ^~~~~~~
In function 'char* strncat(char*, const char*, size_t)',
    inlined from 'void user_term(char*)' at /mgarepo/conky/BUILD/conky-1.18.0/src/users.cc:66:16,
    inlined from 'int update_users()' at /mgarepo/conky/BUILD/conky-1.18.0/src/users.cc:164:12:
/usr/include/bits/string_fortified.h:138:34: error: 'char* __builtin___strncat_chk(char*, const char*, long unsigned int, long unsigned int)' argument 2 declared attribute 'nonstring' [-Werror=stringop-overread]
  138 |   return __builtin___strncat_chk (__dest, __src, __len,
      |          ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
  139 |                                   __glibc_objsize (__dest));
      |                                   ~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/bits/utmp.h: In function 'int update_users()':
/usr/include/bits/utmp.h:62:8: note: argument 'utmp::ut_line' declared here
   62 |   char ut_line[UT_LINESIZE]
      |        ^~~~~~~
cc1plus: all warnings being treated as errors
/mgarepo/conky/BUILD/conky-1.18.0/src/linux.cc: In function 'void get_battery_power_draw(char*, unsigned int, const char*)':
/mgarepo/conky/BUILD/conky-1.18.0/src/linux.cc:2370:10: error: ignoring return value of 'char* fgets(char*, int, FILE*)' declared with attribute 'warn_unused_result' [-Werror=unused-result]
 2370 |     fgets(current_now_val, 256, current_now_file);
      |     ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/mgarepo/conky/BUILD/conky-1.18.0/src/linux.cc:2371:10: error: ignoring return value of 'char* fgets(char*, int, FILE*)' declared with attribute 'warn_unused_result' [-Werror=unused-result]
 2371 |     fgets(voltage_now_val, 256, voltage_now_file);
      |     ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gmake[2]: *** [src/CMakeFiles/conky_core.dir/build.make:725: src/CMakeFiles/conky_core.dir/users.cc.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
cc1plus: all warnings being treated as errors
gmake[2]: *** [src/CMakeFiles/conky_core.dir/build.make:711: src/CMakeFiles/conky_core.dir/linux.cc.o] Error 1
gmake[2]: Leaving directory '/mgarepo/conky/BUILD/conky-1.18.0/build'
gmake[1]: *** [CMakeFiles/Makefile2:443: src/CMakeFiles/conky_core.dir/all] Error 2
gmake[1]: Leaving directory '/mgarepo/conky/BUILD/conky-1.18.0/build'
gmake: *** [Makefile:169: all] Error 2
error: Bad exit status from /mgarepo/conky/BUILDROOT/rpm-tmp.pAzUds (%build)

RPM build errors:
    Bad exit status from /mgarepo/conky/BUILDROOT/rpm-tmp.pAzUds (%build)
error: failed!
```

```shell
gcc --version
gcc (Mageia 12.2.1-0.20230218.1.1.zen3.mga9) 12.2.1 20230218
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

### Version

1.18.0

### Which OS/distro are you seeing the problem on?

Linux (other)

### Conky config

```Lua
N/A
```


### Stack trace

_No response_

### Relevant log output

_No response_

--- a/src/linux.cc
+++ b/src/linux.cc
@@ -2367,14 +2367,14 @@ void get_battery_power_draw(char *buffer, unsigned int n, const char *bat) {
   voltage_now_file = open_file(voltage_now_path, &reported);
 
   if (current_now_file != nullptr && voltage_now_file != nullptr) {
-    fgets(current_now_val, 256, current_now_file);
-    fgets(voltage_now_val, 256, voltage_now_file);
+    if (fgets(current_now_val, 256, current_now_file) &&
+        fgets(voltage_now_val, 256, voltage_now_file)) {
+      current_now = strtol(current_now_val, &ptr, 10);
+      voltage_now = strtol(voltage_now_val, &ptr, 10);
 
-    current_now = strtol(current_now_val, &ptr, 10);
-    voltage_now = strtol(voltage_now_val, &ptr, 10);
-
-    result = (double)(current_now * voltage_now) / (double)1000000000000;
-    snprintf(buffer, n, "%.1f", result);
+      result = (double)(current_now * voltage_now) / (double)1000000000000;
+      snprintf(buffer, n, "%.1f", result);
+    }
     fclose(current_now_file);
     fclose(voltage_now_file);
   }

--- a/src/users.cc
+++ b/src/users.cc
@@ -40,9 +40,8 @@ static void user_name(char *ptr) {
   setutent();
   while ((usr = getutent()) != nullptr) {
     if (usr->ut_type == USER_PROCESS) {
-      if (strlen(ptr) + strnlen(usr->ut_name, UT_NAMESIZE - 1) + 1 <= BUFLEN) {
-        strncat(ptr, usr->ut_name, UT_NAMESIZE);
-      }
+      memcpy(ptr, usr->ut_name, UT_NAMESIZE);
+      ptr[UT_NAMESIZE] = 0;
     }
   }
 }
@@ -62,9 +61,8 @@ static void user_term(char *ptr) {
   setutent();
   while ((usr = getutent()) != nullptr) {
     if (usr->ut_type == USER_PROCESS) {
-      if (strlen(ptr) + strnlen(usr->ut_line, UT_LINESIZE) + 1 <= BUFLEN) {
-        strncat(ptr, usr->ut_line, UT_LINESIZE);
-      }
+      memcpy(ptr, usr->ut_line, UT_LINESIZE);
+      ptr[UT_LINESIZE] = 0;
     }
   }
 }

